src/server/server.ts

```diff
@@ -42,6 +44,66 @@ import { userRouter } from './routes/user';
 // Prompts cache
 let allPrompts: PromptWithMetadata[] | null = null;
 
+export async function checkRemoteHealth(apiUrl: string): Promise<{
+  status: 'OK' | 'ERROR' | 'DISABLED';
+  message: string;
+}> {
+  try {
+    const response = await fetch(apiUrl, {
+      signal: AbortSignal.timeout(1000),
```

Comment:
Would either
1) let it fail open (i.e. don't fail due to timeout)
and/or
2) increase this timeout - I've sometimes seen server hang a little for first request